pipenv/routines/install.py

```diff
@@ -334,8 +334,21 @@ def do_install(
                 )
                 # Add the package to the Pipfile.
                 if index_url:
+                    trusted_hosts = get_trusted_hosts()
+                    host_and_port = get_host_and_port(index_url)
+                    require_valid_https = not any(
```

Comment:
Just thinking through this change -- isn't this saying it requires valid https if any of the hosts in the `PIP_TRUSTED_HOSTS`?  Wouldn't it be better to only check if the specific host of the index_url requires valid https?

Reply:
This should be identical to the code for the requirements file. I assumed it was correct as-is and literally just copied it over. Let me look more into this.